_pytest/config.py

```diff
@@ -106,6 +107,10 @@ def _prepareconfig(args=None, plugins=None):
         if not isinstance(args, str):
             raise ValueError("not a string or argument list: %r" % (args,))
         args = shlex.split(args, posix=sys.platform != "win32")
+        # we want to remove this way of passing arguments to pytest.main()
+        # in pytest-4.0
+        warning = 'passing a string to pytest.main() is deprecated, ' \
+                  'pass a list of arguments instead.'
```

Comment:
I'd really prefer putting parentheses around the string instead of using `\` which can be very brittle (e.g. if someone accidentally adds a space after it)